fake_joint_driver/src/fake_joint_driver_node.cpp

```diff
@@ -6,36 +6,62 @@
  *
  * Device driver node to fake loopback joints.
  */
-#include "ros/ros.h"
-#include "controller_manager/controller_manager.h"
-#include "fake_joint_driver/fake_joint_driver.h"
+#include <controller_manager/controller_manager.hpp>
+#include <fake_joint_driver/fake_joint_driver.h>
+#include <rclcpp/rclcpp.hpp>
+
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("fake_joint_driver");
+
+void spin(std::shared_ptr<rclcpp::executors::MultiThreadedExecutor> exe)
+{
+  exe->spin();
+}
 
 /**
  * @brief Main function
  */
-int main(int argc, char **argv)
-{
+int main(int argc, char **argv) {
   // Init ROS node
-  ros::init(argc, argv, "fake_joint_driver");
-  ros::NodeHandle nh;
-
+  rclcpp::init(argc, argv);
+  auto executor = std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
   // Create hardware interface
-  FakeJointDriver robot;
+  auto robot = std::make_shared<FakeJointDriver>(rclcpp::Node::make_shared("fake_joint_driver_node"));
+
   // Connect to controller manager
-  controller_manager::ControllerManager cm(&robot, nh);
+  controller_manager::ControllerManager cm(robot, executor);
 
-  // Set spin ratge
-  ros::Rate rate(1.0 / ros::Duration(0.010).toSec());
-  ros::AsyncSpinner spinner(1);
-  spinner.start();
+  cm.load_controller("ros_controllers", "ros_controllers::JointStateController", "fake_joint_state_controller");
+  cm.load_controller("ros_controllers", "ros_controllers::JointTrajectoryController",
+                     "fake_joint_trajectory_controller");
+
+  // there is no async spinner in ROS 2, so we have to put the spin() in its own thread
+  auto future_handle = std::async(std::launch::async, spin, executor);
```

Comment:
We have no use for the future. Simply run `executor->spin()` in a thread using a lambda function